pkg/app/app_reconcile.go

```diff
@@ -60,7 +60,19 @@ func (a *App) reconcileDelete() error {
 
 	a.resetLastDeployStartedAt()
 
-	result := a.delete(a.updateLastDeployNoReturn)
+	var result exec.CmdRunResult
+	if !a.app.Spec.NoopDelete {
+		result = a.delete(a.updateLastDeployNoReturn)
+	} else {
+		// If NoopDelete, preserve App's associated resources
+		// but delete App CR itself by removing finalizer.
+		err = a.unblockDeletion()
+		if err != nil {
+			result = exec.CmdRunResult{
+				Error: err,
+			}
+		}
```

Comment:
this code is also present inside delete(). it would be good if it only exists in one place.

Reply:
Moved the logic into `delete()`